fix(frontend): enhance baodaozai visibility management

ericsen-tsai opened the pull request:
## Summary

This PR improves the Baodaozai component's visibility management by ensuring it starts in a hidden state until properly initialized, and adds proper cleanup when visibility control components unmount. This prevents the component from briefly flashing visible before initialization completes.

## Changes

- Added cleanup function in `BaodaozaiVisibilitySetter` component to reset hide state to `false` when component unmounts
- Introduced `innerHide` state in `Baodaozai` component initialized to `true` to ensure component starts invisible
- Added `useEffect` hook to synchronize `innerHide` state with the `hide` prop
- Updated opacity class logic to use `innerHide` instead of directly using the `hide` prop

## Additional Notes

The changes ensure that:
- Baodaozai remains invisible during initial render until the visibility state is properly set
- When `BaodaozaiVisibilitySetter` unmounts, the hide state is reset to `false`, returning Baodaozai to its default visible state
- The component avoids visual glitches from appearing before initialization completes

## Screenshot(s)

https://github.com/user-attachments/assets/a84b1c74-277a-49d8-9a3b-dc383a4beb13




## Reference(s)

- [Notion Page](https://www.notion.so/twreporter/defect-2a88dbdca9328074a39fc25135b00285?v=1588dbdca93281dfa43c000c8415265e&source=copy_link)

ericsen-tsai committed:
fix(frontend): enhance baodaozai visibility management

- Added cleanup logic to reset the hide state when the BaodaozaiVisibilitySetter component unmounts.
- Introduced innerHide state in the Baodaozai component to manage visibility during initialization.

nickhsine commented on packages/frontend/src/services/call-baodaozai/components/baodaozai.tsx:
```diff
@@ -69,11 +69,18 @@ function Baodaozai() {
+
+  useEffect(() => {
+    setInnerHide(hide)
+  }, [hide])
```
thought:
是不是讓 `hide` 預設是 true 呢？
這樣子是不是不需要再多一個 innerHide state？

ericsen-tsai replied:
I changed hide to "true" by default.
[fix(frontend): mark isExpanded as transient props to prevent warning](https://github.com/kids-reporter/kids-reporter-monorepo/pull/827/commits/bf99e80f6e57e4fe6ef983a7f1e3c96c18d9d696)

nickhsine commented on packages/frontend/src/services/call-baodaozai/components/baodaozai-visibility-setter.tsx:
```diff
@@ -9,6 +9,11 @@ function BaodaozaiVisibilitySetter({ show }: { show: boolean }) {
+    return () => {
+      // reset hide state when component unmounts
+      setHide(false)
+    }
```
question:
目前 `BaodaozaiVisibilitySetter` 僅用在 article 頁面上，如果 article 頁切到另一個 article 頁面時，會觸發 unmounts 嗎？

ericsen-tsai replied:
I removed this logic to prevent unexpected behavior between route changes.
[fix(frontend): mark isExpanded as transient props to prevent warning](https://github.com/kids-reporter/kids-reporter-monorepo/pull/827/commits/bf99e80f6e57e4fe6ef983a7f1e3c96c18d9d696)

ericsen-tsai committed:
fix(frontend): update Baodaozai visibility management across multiple components

- Added BaodaozaiVisibilitySetter to various pages to control visibility.
- Refactored Baodaozai component to initialize visibility state correctly.
- Removed cleanup logic from BaodaozaiVisibilitySetter to simplify state management.

ericsen-tsai committed:
fix(frontend): mark isExpanded as transient props to prevent warning

ericsen-tsai commented:
@nickhsine 
There's one extra fix for the styled component props warning.

<img width="1416" height="97" alt="image" src="https://github.com/user-attachments/assets/f5fe4151-1edc-4694-b20a-92ba31beb2db" />

[fix(frontend): mark isExpanded as transient props to prevent warning](https://github.com/kids-reporter/kids-reporter-monorepo/pull/827/commits/bf99e80f6e57e4fe6ef983a7f1e3c96c18d9d696)

nickhsine commented:
LGTM 👍 👍
